static/js/impala/suggestions.js

```diff
@@ -130,12 +130,14 @@ $.fn.searchSuggestions = function($results, processCallback, searchType) {
             $results.filter('.visible').removeClass('visible');
             return;
         }
+        var urlVal = encodeURIComponent($self.val());
```

Comment:
> Seems, it happens on JS Bin only. Try escape_(encodeURIComponent("'foo'")) in browser's console.

It's not just the escaping and encoding you need to append the HTML into the DOM.